run_tree/data/shaders/shaders.hlsl

```diff
@@ -0,0 +1,39 @@
+cbuffer constants : register(b0) {
+  column_major float4x4 transform;
+  column_major float4x4 projection;
+}
+
+/* vertex attributes go here to input to the vertex shader */
+struct vs_in {
+  float2 position_local : POS;
+  uint   packed_color   : COL;
+};
+
+/* outputs from vertex shader go here. can be interpolated to pixel shader */
+struct vs_out {
+  float4 position_clip : SV_POSITION; // required output of VS
+  float4 color         : COL;
+};
+
+float4 unpack_color(uint packedColor) {
+  float4 color;
+  color.r = (float)((packedColor >> 24) & 0xFF) / 255.0;
+  color.g = (float)((packedColor >> 16) & 0xFF) / 255.0;
+  color.b = (float)((packedColor >> 8)  & 0xFF) / 255.0;
+  color.a = (float)( packedColor        & 0xFF) / 255.0;
+  return color;
+}
```

Comment:
Packing color in meshes is not needed, memory savings are negligible while computation costs is still there (all buffers are allocated on 64 KB granularity in D3D anyway, only difference is dynamic constant buffers from DX11 but they live in UPLOAD heap and are meant for per frame/pass/draw data).